python/ray/data/_internal/execution/interfaces/physical_operator.py

```diff
@@ -188,6 +188,9 @@ def __init__(
         self._estimated_num_output_bundles = None
         self._estimated_output_num_rows = None
         self._execution_completed = False
+        # The LogicalOperator which was translated to create this PhysicalOperator.
+        # Set via `PhysicalOperator.set_logical_operator()`.
+        self._logical_op = None
```

Comment:
this should be a list for fused physical operators.

Reply:
good point. i was looking at this, and realized that during operator fusion, we actually construct a new `LogicalOperator` to represent the fused `PhysicalOperator`s. For example for fusing two Map->Map ops, we construct a new logical Map op: https://github.com/ray-project/ray/blob/master/python/ray/data/_internal/logical/rules/operator_fusion.py#L355-L360

so i was thinking we should link this newly constructed logical op, instead of combining the list of linked logical ops from the physical operators being fused, since this most closely reflects what's going on under the hood. let me know if you think we should still maintain the full list of logical operators (i can still see that being useful for keeping track of where linked logical operators end up).